### PR TITLE
adding functionality for Youtube

### PR DESCRIPTION
**Pull request by royboy789**

### CODE REVIEW

File: cf-video-overlays.php [hunk 1]
```diff
@@ -47,6 +46,14 @@ public function __construct(){
 		add_filter( 'shortcode_atts_caldera_form', array( $this, 'shortcode_atts' ), 10, 4 );
 		add_filter( 'shortcode_atts_caldera_forms_modal', array( $this, 'shortcode_atts' ), 10, 4 );
 		add_filter( 'caldera_forms_pre_render_form', array( $this, 'maybe_load' ), 10, 4 );
+		add_filter( 'cf_video_overlays_allowed_sources', [ $this, 'allowed_sources'] );
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'load_scripts' ) );
+	}
+
+	function allowed_sources( $allowed ) {
```
Comment (Shelob9) [inline cf-video-overlays.php]: @royboy789 This method is lacking inline docs and a visibility declaration

File: cf-video-overlays.php [hunk 2]
```diff
@@ -24,10 +24,9 @@ function cf_video_overlays_init(){
  *
  * @return bool
  */
-function cf_video_overlays_allowed_source( $source_type){
-	$allowed = apply_filters( 'cf_video_overlays_allowed_sources', array(
-		'youtube'
-	) );
+function cf_video_overlays_allowed_source( $source_type ){
+	$allowed = [];
```
Comment (Shelob9) [inline cf-video-overlays.php]: @royboy789 Let's keep this to PHP5.3 compat please

File: cf-video-overlays.php [hunk 3]
```diff
@@ -47,6 +46,14 @@ public function __construct(){
 		add_filter( 'shortcode_atts_caldera_form', array( $this, 'shortcode_atts' ), 10, 4 );
 		add_filter( 'shortcode_atts_caldera_forms_modal', array( $this, 'shortcode_atts' ), 10, 4 );
 		add_filter( 'caldera_forms_pre_render_form', array( $this, 'maybe_load' ), 10, 4 );
+		add_filter( 'cf_video_overlays_allowed_sources', [ $this, 'allowed_sources'] );
```
Comment (Shelob9) [inline cf-video-overlays.php]: @royboy789 Let's keep this to PHP5.3 compat please

File: cf-video-overlays.php [hunk 4]
```diff
@@ -47,6 +46,14 @@ public function __construct(){
...
+
```
Comment (Shelob9) [inline cf-video-overlays.php]: @royboy789  This extra line is uneeded

File: video-templates/scripts/cf-source-youtube.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+var $ = jQuery,
+    video_container = $('#cf_youtube_overlay_container'),
```
Comment (Shelob9) [inline video-templates/scripts/cf-source-youtube.js]: @royboy789 for consitency with other Caldera code, please use $video_container I always use $name for a jQuery object in a variable

File: video-templates/scripts/cf-source-youtube.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
+var $ = jQuery,
```
Comment (Shelob9) [inline video-templates/scripts/cf-source-youtube.js:1]: @royboy789 Why this versus noconflict wrapper? Just curious, assuming I'm right that var declaration keeps in scope of this file, and therefore you're not messing with $ in other scopes, then whatever, just curious.

File: video-templates/styles/cf-video-overlay-styles.css [hunk 7]
```diff
@@ -0,0 +1,50 @@
+/** Responsive **/
+.embed-responsive {
```
Comment (Shelob9) [inline video-templates/styles/cf-video-overlay-styles.css]: @royboy789 Please prefix all CSS classes with .cf_video_overlays or use .cf_video_overlays .class-name These classes are likely to be used in other plugins, themes, etc.

File: video-templates/youtube.php [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?php
+$output = '<div id="cf_youtube_overlay_wrapper" class="embed-responsive embed-responsive-16by9">';
```
Comment (Shelob9) [inline video-templates/youtube.php]: @royboy789  Please put form ID in a class for overlay specific styling.

File: video-templates/scripts/cf-source-youtube.js (hunk 6 above)
Comment (royboy789) [reply video-templates/scripts/cf-source-youtube.js:1]: line 19, the onYoutubeIframeApiReady... its really fucking finnicky.

**Pull request merged**